bases/renku_data_services/background_jobs/main.py

```diff
@@ -23,21 +24,39 @@ async def short_period_sync() -> None:
     """Perform synchronizations and jobs that should occur more often."""
     config = SyncConfig.from_env()
     run_migrations_for_app("common")
-    await bootstrap_user_namespaces(config)
-    await config.syncer.events_sync(config.kc_api)
-    await sync_admins_from_keycloak(config.kc_api, Authz(config.authz_config))
-    await fix_mismatched_project_namespace_ids(config)
-    await migrate_groups_make_all_public(config)
-    await migrate_user_namespaces_make_all_public(config)
-    await migrate_storages_v2_to_data_connectors(config)
+
+    error_handler = ErrorHandler()
+
+    async with error_handler:
+        await bootstrap_user_namespaces(config)
+    async with error_handler:
+        await config.syncer.events_sync(config.kc_api)
+    async with error_handler:
+        await sync_admins_from_keycloak(config.kc_api, Authz(config.authz_config))
+    async with error_handler:
+        await fix_mismatched_project_namespace_ids(config)
+    async with error_handler:
+        await migrate_groups_make_all_public(config)
+    async with error_handler:
+        await migrate_user_namespaces_make_all_public(config)
+    async with error_handler:
+        await migrate_storages_v2_to_data_connectors(config)
```

Comment:
I wonder if we could make this a bit more elegant.
```python
promises = [
  bootstrap_user_namespaces(config),
  config.syncer.events_sync(config.kc_api),
  ...
]

for promise in promises:
  async with error_handler:
    await promise
```

Reply:
I originally considered ding a loop but then thought that there could be cases where job2 shouldn't run if job1 fails, which would be
```
async with error_handler:
    await job1()
    await job2()
```

You could achieve the same in a loop probably by composing the jobs or having separate loops. But I thought the straight forward but repetitive way is the easiest to understand.

In the end,m this is essentially a monad, so I can look a bit if there's an elegant way of having monads in python.

Reply:
It was a non-blocking comment anyway. I can try to make a PR later.